lib/sha256.c

```diff
@@ -43,7 +43,7 @@
 #include <mbedtls/version.h>
 
 #if(MBEDTLS_VERSION_NUMBER >= 0x02070000)
-  #define HAS_RESULT_CODE_BASED_FUNCTIONS
+  #define HAS_MBEDTLS_RESULT_CODE_BASED_FUNCTIONS
```

Comment:
Wouldn't the code be easier if this code block instead was made like?
~~~c
#if(MBEDTLS_VERSION_NUMBER >= 0x02070000) && \
   (MBEDTLS_VERSION_NUMBER < 0x03000000)
  #define HAS_MBEDTLS_RESULT_CODE_BASED_FUNCTIONS
#endif